src/utils/slateHtmlSerializer.js

```diff
@@ -39,6 +39,22 @@ const getIframeType = (src) => {
   return null;
 };
 
+/**
+ * Return an object of target attribute values, and will return all attribute values of the node if input is empty.
+ * @param {Element} node The html node
+ * @param {...string} attrs Attribute names
+ * @returns {Object.<string, string>} An object of attr values.
+ */
+const getNodeAttrs = function(node, ...attrs){
+  return (attrs.length ? attrs : node.getAttributeNames()).reduce((obj, attr) => {
+    let name = attr;
+    if(attr === 'class') name = 'className';
+
+    obj[name] = node.getAttribute(attr);
+    return obj;
+  }, {})
```

Comment:
NIT：這邊有兩個小建議～
1. `(attrs.length ? attrs : node.getAttributeNames()).reduce(/* ... */)` 可以把 `attributeNames` 抽出成變數，會比較好閱讀，因為兩個或兩個以上的 expression 混在一起會比較不容易一眼就理解
2. `attributeName` 可以直接用 Ternary Operator 取代 `if` condition
3. `obj` 可改用 `attributes` 會更符合語意
4. reducer 儘量為 pure，儘量不要用 mutation 的方式修改 `obj`，回傳值可把前一次結果 Destructuring Assignment 給一個新的 result

```suggestion
  const attributeNames = attrs.length > 0 ? attrs : node.getAttributeNames();
  
  return attributeNames.reduce(
    (attributes, attr) => {
      const attributeName = attr === 'class' ? 'className' : attr;
      const attributeValue = node.getAttribute(attr);

      return { ...attributes, [attributeName]: attributeValue };
    },
    {}
  );
```

Reply:
`if(attr === 'class') name = 'className';` 是預留給 attribute 在 react 有不同名稱時的彈性，第二點先暫時不予更動

Reply:
想問問第四點保持 immutable 優點？

Reply:
> `if(attr === 'class') name = 'className';` 是預留給 attribute 在 react 有不同名稱時的彈性，第二點先暫時不予更動

也許可以在上面加個 comment

> 想問問第四點保持 immutable 優點？

讓 reducer 為 pure function 可以很清楚的知道 loop 時每次迭代之間的差異，要找問題的時候會容易許多

Reply:
上次跟你聊完本來想說寫個註解，但最後還是改用 object mapping 了，這樣未來擴增實也比較清楚

第四點我再想一下，但因為這邊沒有太大的差異，所以就先維持原方案～

9fb8464